src/main/java/ResourceDiscovery/AssetObjects/AssetObject.java

```diff
@@ -1,49 +1,46 @@
 package ResourceDiscovery.AssetObjects;
 
+import ResourceDiscovery.AssetTypes;
 import ResourceDiscovery.ProjectObjects.ProjectConfig;
+import com.google.cloud.Timestamp;
 import com.google.common.flogger.FluentLogger;
-import org.springframework.cloud.gcp.data.spanner.core.mapping.Column;
-import org.springframework.cloud.gcp.data.spanner.core.mapping.NotMapped;
-import org.springframework.cloud.gcp.data.spanner.core.mapping.PrimaryKey;
-import org.springframework.cloud.gcp.data.spanner.core.mapping.Table;
 
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
-import java.util.Date;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-@Table(name = "Assets")
+/**
+ * The AssetObject class is an abstract class which is the parent of all of the specific asset
+ * object classes. This class provides setters and getters to all of the common fields of the
+ * asset objects. In addition it provides some helper functions for specific type parsing and
+ * conversions which are used in several asset object classes (such as a convertStringToDate function).
+ */
 abstract public class AssetObject {
-    @PrimaryKey(keyOrder = 1)
-    private String accountId = ProjectConfig.getInstance().getAccountId();
-    @PrimaryKey(keyOrder = 2)
-    private String projectId = ProjectConfig.getInstance().getProjectId();
-    // We are using both the asset name and kind in order to ensure uniqueness
-    @PrimaryKey(keyOrder = 4)
+    // asset primary keys
+    public String accountId = ProjectConfig.getInstance().getAccountId();
+    public String projectId = ProjectConfig.getInstance().getProjectId();
     protected String kind;
-    @PrimaryKey(keyOrder = 3)
-    @Column(name = "assetName")
     protected String name;
 
-    @Column(name = "assetId")
+    // asset additional data
     protected String id;
-    @Column(name = "assetType")
     protected String type;
     protected String zone;
-    protected Date creationTime;
+    protected Timestamp creationTime;
     protected String status;
 
-    @NotMapped
+    protected AssetTypes assetTypeEnum;
```

Comment:
Two naming comments:
1. I actually don't see the definition here, but I think it's more common to define enum type-names as singular "AssetType"
2. Unless there is a risk of confusing this with some other asset-type, I think this can simply be called "assetType" and the readers can find that it's an enum themselves. (unless I'm missing some other consideration)

Also many places below

Reply:
Regarding the second comment - in the asset table in our spanner DB the "type" column is called "assetType" and that was the reason for the more descriptive name "assetTypeEnum". The logic behind the "type" and "assetType" inconsistency was that when reading from the asset tables it might be important for the column names to have more meaningful names rather than the fields in the asset objects which are obviously asset related. Maybe this inconsistency should be changed (for either more fully descriptive names in our code or less in our tables), what do you think?

Reply:
I'm not sure I fully follow... Do the 'assetTypeEnum' and the 'type' fields in Java represent the same field in the DB?

Reply:
No, the "type" field is something returned from the assets (for example for an asset of Cloud SQL Instance it can be something like "mysql") and the assetTypeEnum does not have a field in the DB - its a constant used to decide which specific asset type should be created. 
Perhaps we can discuss this shortly tomorrow in our tri-weekly as I think it effects something a bit bigger then this in the design